packages/grafana-ui/src/components/TimeSeries/utils.ts

```diff
@@ -217,9 +215,6 @@ export const preparePlotConfigBuilder: UPlotConfigPrepFn<{
       continue;
     }
 
-    // TODO: skip this for fields with custom renderers?
-    field.state!.seriesIndex = seriesIndex++;
-
     let fmt = field.display ?? defaultFormatter;
```

Comment:
This file looks important 👀 Do all panels based on uplot have util functions to generate uplot config? Also why is this located in grafana-ui vs core?

Reply:
added a test for `prepareGraphableFields()` to make sure it enumerates this properly. i checked references for these prepareGraphableFields and preparePlotConfigBuilder and it looks fully isolated to TimeSeries.

Candlestick (which is built on top of TimeSeries) also uses prepareGraphableFields():

https://github.com/grafana/grafana/blob/f561e71de87bcfdbc07ad4dd1af11727d03431b8/public/app/plugins/panel/candlestick/fields.ts#L124

and then re-enumerates the seriesIndex anyhow:

https://github.com/grafana/grafana/blob/f561e71de87bcfdbc07ad4dd1af11727d03431b8/public/app/plugins/panel/candlestick/fields.ts#L239-L254

so i think that one should be okay.

Reply:
> why is this located in grafana-ui vs core?

because moving it is too hard right now 😬  It got started there thinking it would be a reasonable way to expose basic components to plugins, but "basic" shifts to super super specific quickly and this really should be internal API's and components.